jgrapht-core/src/test/java/org/jgrapht/alg/color/Bipartite2ColoringTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * (C) Copyright 2018-2018, by Meghana M Reddy and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.color;
+
+import java.util.Map;
+
+import org.jgrapht.Graph;
+import org.jgrapht.alg.interfaces.VertexColoringAlgorithm.Coloring;
+import org.jgrapht.graph.DefaultDirectedGraph;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.Pseudograph;
+import org.junit.*;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test for the BipartiteColoring class.
+ * 
+ * @author Meghana M Reddy
+ *
+ */
+public class Bipartite2ColoringTest {
+    public void test1() {
```

Comment:
Can you add a few lines of text, explaining what you are testing? E.g. "test on non-bipartite graph" "test on disconnected, directed bipartite graph" etc. That makes it easier to assess what kind of test cases you covered without having to draw the graphs.

Reply:
I have added additional test cases and their corresponding descriptions.